pkg/configuration/digger_config.go

```diff
@@ -388,6 +407,41 @@ func (c *DiggerConfig) GetWorkflowConfiguration(projectName string) WorkflowConf
 	return *workflow.Configuration
 }
 
+func checkSteps(steps []Step) error {
```

Comment:
I think that the checks in this function are incorrect because it should be checking that for a plan flow: at least `init` and `plan` exists

for apply it should check that at least `init` and `apply` flow exists

Reply:
do you means, the plan and apply need difference checking, e.g.
func checkPlanSteps(steps []Step) error {
	for _, step := range steps {
		if (step.Action == "init" || step.Action == "plan") && (step.Value == "" && step.ExtraArgs == nil) {
			return fmt.Errorf("%s", step.Action)
		}
	}

	return nil
}

func checkApplySteps(steps []Step) error {
	for _, step := range steps {
		if (step.Action == "init" || step.Action == "apply") && (step.Value == "" && step.ExtraArgs == nil) {
			return fmt.Errorf("%s", step.Action)
		}
	}

	return nil
}

func (c *DiggerConfig) CheckWorkflowConfiguration() error {
	workflows := c.Workflows

	for _, project := range c.Projects {
		workflow, ok := workflows[project.Workflow]

		if !ok {
			return fmt.Errorf("project '%s' specifies workflow not in list of workflows", project.Name)
		}

		err := checkPlanSteps(workflow.Plan.Steps)
		if err != nil {
			return fmt.Errorf(ErrDiggerInvalidWorkflow.Error(), project.Workflow, err.Error())
		}

		err = checkApplySteps(workflow.Apply.Steps)
		if err != nil {
			return fmt.Errorf(ErrDiggerInvalidWorkflow.Error(), project.Workflow, err.Error())
		}
	}

	return nil
}

Reply:
No its more like that for every workflow we need to check that there is BOTH init and plan. In your function the check is only for a single check. The logic is like this:

```
for all workflows {
     check if there is a plan step then {
            check if both init and plan steps exist
            if they don't exist then throw error
     }
    check if there is apply step then {
         if apply step exists then both init and apply steps exist
    }
}
```

Hope that helps

Reply:
thanks, I got.

and I noticed, there are some unit tests, e.g. "TestDiggerConfigWhenOnlyYamlExists", is it invalid when we apply new checking? 
func TestDiggerConfigWhenOnlyYamlExists(t *testing.T) {
	tempDir, teardown := setUp()
	defer teardown()

	diggerCfg := `
projects:
- name: prod
  branch: /main/
  dir: path/to/module/test
  workspace: default
`
	deleteFile := createFile(path.Join(tempDir, "digger.yaml"), diggerCfg)
	defer deleteFile()

	dg, err := NewDiggerConfig(tempDir, &FileSystemDirWalker{})
	assert.NoError(t, err, "expected error to be nil")
	assert.NotNil(t, dg, "expected digger config to be not nil")
	assert.Equal(t, "path/to/module/test", dg.GetDirectory("prod"))
}